api/src/main/java/org/openmrs/obs/handler/ImageHandler.java

```diff
@@ -96,7 +96,15 @@ public Obs getObs(Obs obs, String view) {
 				// Do not fail if image is missing
 			}
 			
-			ComplexData complexData = new ComplexData(key, img);
+			String filename = parseFilename(obs, "image");
+			
+			// hack--if this is old version of the image where we aren't explicitly saving the filename, we
+			// need to use "key"/"path" portion as the file, see https://openmrs.atlassian.net/browse/TRUNK-6472
+			if (extensions.contains(filename)) {
+				filename = key.replaceFirst(getObsDir(),"");
+			}
+			
```

Comment:
Hack to make sure we can still read the "old" format.  We consider it the "old" format if the "filename" is *just* one of our supported extension types.